node/client/src/benchmarking.rs

```diff
@@ -368,8 +368,28 @@ pub fn benchmark_inherent_data(
 		parent_header: header,
 	};
 
-	polkadot_node_core_parachains_inherent::ParachainsInherentDataProvider::from_data(para_data)
-		.provide_inherent_data(&mut inherent_data)?;
+	struct BenchmarkProvider(polkadot_primitives::v2::InherentData);
+	#[async_trait::async_trait]
+	impl sp_inherents::InherentDataProvider for BenchmarkProvider {
+		async fn provide_inherent_data(
+			&self,
+			dst_inherent_data: &mut sp_inherents::InherentData,
+		) -> Result<(), sp_inherents::Error> {
+			dst_inherent_data.put_data(*b"BENCH_PR", &self.0)
```

Comment:
This is wrong. You need to use the correct `ParachainsInherentDataProvider`. And if you are re-implementing the `InherentDataProvider` any way, you can don't need to do it. Just create the `InherentData` manually.

Reply:
I'm not following. I asked if I should do what I did https://github.com/paritytech/polkadot/pull/6328#discussion_r1032602742 and you said I can. What was my misunderstanding?

At the end of the day, what exactly is this supposed to benchmark? `provide_inherent_data(...)` for `ParachainsInherentDataProvider`? `dst_inherent_data.put_data(...)` for the default `ParachainsInherentData`?

Reply:
The first parameter to `put_data` is a unique identifier. If you change this identifier, the runtime functionality will not find this inherent data.

However, I just checked the code and we don't really need this.  @ggwpez can chime in to support my assumption here, but we probably use this to build some empty block for benchmarking. This empty block would also work without providing any inherent data.

TLDR, you should be able to remove the entire code here.

Reply:
I think the `paras-inheret` pallet requires `PARACHAINS_INHERENT_IDENTIFIER` to be present, but it can be anything.  
So something like this should be enough:  
```rust
inherent_data.put_data(polkadot_primitives::v2::PARACHAINS_INHERENT_IDENTIFIER, &para_data);
```

I cannot compile the code even after diener but you can test it with `benchmark overhead --dev --max-ext-per-block 10` sub-command.